tests/unit/ssg-module/test_templates.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+import ssg.templates as tpl
+from ssg.environment import open_environment
+import ssg.utils
+import ssg.products
+from ssg.yaml import ordered_load
+import ssg.build_yaml
+import ssg.build_cpe
+
+
+ssg_root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))
+DATADIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "data"))
+templates_dir = os.path.join(DATADIR, "templates")
+cpe_items_dir = os.path.join(DATADIR, "applicability")
+
+build_config_yaml = os.path.join(ssg_root, "build", "build_config.yml")
+product_yaml = os.path.join(ssg_root, "products", "rhel8", "product.yml")
+env_yaml = open_environment(build_config_yaml, product_yaml)
+
+
+def test_render_extra_ovals():
+    builder = ssg.templates.Builder(
+        env_yaml, '', templates_dir,
+        '', '', '', cpe_items_dir)
+
+    declaration_path = os.path.join(builder.templates_dir, "extra_ovals.yml")
+    declaration = ssg.yaml.open_raw(declaration_path)
+    for oval_def_id, template in declaration.items():
+        oval_content = builder.get_lang_contents(oval_def_id, oval_def_id, template,
+                                                 ssg.templates.languages["oval"])
+        assert "<title>%s</title>" % (oval_def_id,) in oval_content
```

Comment:
I'm sure that @matejak would not agree with you on this.

But regardless it is just an introduction of tests for template building, they will evolve in consecutive PRs.

Reply:
Indeed, you don't know how the output should *exactly* look like, you don't want to write tests that are too specific, because those tests would be fragile. As the template code can be subject of change, you want to test only selected general assumptions to make sure that the ends meet.
This is different from testing XCCDF output, as there is only one right way how to represent certain data in the XCCDF format, and in such case, the test comparing the generated XML against a reference isn't fragile.

Reply:
I'd simply say that in a unit test one should focus on testing things that are coming via function arguments and avoid testing side-effects at all costs.

Reply:
So there shall be other tests that would catch proper ID, product, etc. This particular test at this moment ensures that the title is propagated.